embulk-core/src/main/java/org/embulk/exec/PreviewExecutor.java

```diff
@@ -174,10 +176,15 @@ public void add(Page page)
         @Override
         public void finish()
         {
+            if (res != null) {
+                // if PreviewResult object already exists, we should reuse it and throw PreviewedNoticeError.
```

Comment:
@dmikurube I also thought, that's consistent behavior. Ok, I will revert it and just put logging that I mentioned above.